primitives/src/traits/dispute_api.rs

```diff
@@ -83,10 +89,20 @@ pub trait DisputeApi {
     /// was unable to come to a conclusion.
     /// **May** assume that `market.dispute_mechanism` refers to the calling dispute API.
     fn has_failed(
-        disputes: &[MarketDispute<Self::AccountId, Self::BlockNumber>],
         market_id: &Self::MarketId,
         market: &MarketOfDisputeApi<Self>,
     ) -> Result<bool, DispatchError>;
+
+    /// Called, when a global dispute is started.
+    /// **May** assume that `market.dispute_mechanism` refers to the calling dispute API.
+    fn on_global_dispute(
+        market_id: &Self::MarketId,
+        market: &MarketOfDisputeApi<Self>,
+    ) -> DispatchResult;
+
+    /// Called, when a market is destroyed.
```

Comment:
```
    /// Allow the API consumer to clear storage items of the dispute mechanism.
    /// This may be called, when the dispute mechanism is no longer needed.
    ```

Reply:
c3c5ca6baee310aa05a7c99758db614d372f988c

Reply:
But this doesn't clear the entire storage of the MDM, only the storage related to this particular market.

Reply:
Live example of the current court implementation: https://github.com/zeitgeistpm/zeitgeist/blob/d3f55ace6b81e2d23d73bb660f0c8fcd6d8e6a1e/zrml/court/src/lib.rs#L1503-L1513

Reply:
The clearing is triggered by the prediction markets pallet. So we need to clear all related MDM storage too.